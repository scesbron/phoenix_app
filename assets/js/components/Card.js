import React from "react";
import PropTypes from "prop-types";

class Card extends React.Component {
  render() {
    const { isDragging, title, message } = this.props;
    return (
      <div className="card" style={{ opacity: isDragging ? "0.5" : "1" }}>
        <div className="card-header">
          <div className="card-header-title">{title}</div>
        </div>
        <div className="card-content">{message}</div>
      </div>
    );
  }
}

Card.propTypes = {
  isDragging: PropTypes.bool,
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired
};

export default Card;
