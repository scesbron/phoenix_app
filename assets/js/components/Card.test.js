import React from "react";
import Card from "./Card";
import renderer from "react-test-renderer";

it("renders correctly when not dragging", () => {
  const tree = renderer
    .create(<Card isDragging={false} title="Title" message="Message" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when dragging", () => {
  const tree = renderer.create(<Card isDragging={true} title="Title" message="Message" />).toJSON();
  expect(tree).toMatchSnapshot();
});
