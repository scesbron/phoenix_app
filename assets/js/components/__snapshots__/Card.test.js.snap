// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when dragging 1`] = `
<div
  className="card"
  style={
    Object {
      "opacity": "0.5",
    }
  }
>
  <div
    className="card-header"
  >
    <div
      className="card-header-title"
    >
      Title
    </div>
  </div>
  <div
    className="card-content"
  >
    Message
  </div>
</div>
`;

exports[`renders correctly when not dragging 1`] = `
<div
  className="card"
  style={
    Object {
      "opacity": "1",
    }
  }
>
  <div
    className="card-header"
  >
    <div
      className="card-header-title"
    >
      Title
    </div>
  </div>
  <div
    className="card-content"
  >
    Message
  </div>
</div>
`;
